mindmeld/components/custom_action.py

```diff
@@ -0,0 +1,187 @@
+import logging
+
+import aiohttp
+import requests
+
+from .dialogue import DialogueResponder
+
+
+logger = logging.getLogger(__name__)
+
+
+RESPONSE_FIELDS = ["frame", "directives", "params"]
+
+
+class CustomActionException(Exception):
+    pass
+
+
+class CustomAction:
+    """
+    This class allows the client to send Request and Responder to another server and return the
+      the directives and frame in the response.
+    """
+
+    def __init__(self, name: str, config: dict, overwrite: bool = False):
+        self._name = name
+        self._config = config or {}
+        self.url = self._config.get("url")
+        self.overwrite = overwrite
+
+    def get_json_payload(self, request, responder):
+        request_json = {
+            "text": request.text,
+            "domain": request.domain,
+            "intent": request.intent,
+            "context": dict(request.context),
+            "params": request.params.to_dict(),
+            "frame": dict(request.frame),
+        }
```

Comment:
Lets just make a `to_dict` function for the `Request` class so that we dont have to do this everytime.

Reply:
Done